### PR TITLE
Update select-to-end-of-line behavior

### PR DESCRIPTION
**Pull request by cinnamonKale**
Original PR is in atom/atom#12117, but I borked my local repo and started over. Only after I did that did I learn how to fix it :persevere:

Addresses atom#7159, where `editor:select-to-end-of-line` should behave similarly to `editor:move-to-end-of-line`. This PR does the following to make functions behave as expected and provide functions for existing behaviors (all select/move functions do so from the cursor position to the specified position):
- `editor:select-to-beginning-of-line`: **Unchanged,** selects to the beginning of the buffer line
- `editor:select-to-beginning-of-screen-line`: **New,** selects to the beginning of the screen line
- `editor:select-to-end-of-line`: **Changed,** selects to the end of the buffer line, originally selected to the end of the screen line
- `editor:select-to-end-of-screen-line`: **New,** selects to the end of the screen line, original behavior of `editor:select-to-end-of-line`
- `editor:select-to-first-character-of-line`: **Changed,** selects to the first non-whitespace character of the buffer line, originally selected to the first non-whitespace character of the screen line
- `editor:select-to-first-character-of-screen-line`: **New,** selects to the first non-whitespace character of the screen line, same behavior as original `select-to-first-character-of-line`
- `editor:move-to-first-character-of-line`: **Changed,** moves the cursor to the first non-whitespace character of the buffer line, originally moved to the first non-whitespace character of the screen line
- `editor:move-to-first-character-of-screen-line`: **New,** moves the cursor to the first non-whitespace character of the screen line, original behavior of `editor:move-to-first-character-of-line`

@lee-dohm, I expanded your [original checklist](https://github.com/atom/atom/pull/12117#issuecomment-232172717) to include the first-character functions because it followed the same logic as the other function changes. Admittedly, I've never written specs before, but I'm familiar with Jasmine. Any guidance for writing specs for this would be super helpful.
### TODO:
- [ ] Write specs
- [ ] Cleanup, including old commented code and extra space around new code
- [ ] Decide if the `home` and `end` keys should act on the screen or buffer line. Currently it's the screen line
- [ ] Modify new screen line functions in `menus/linux.cson` and `menus/win32.cson` to have shortcuts (at lease I think that's what the ampersand is for on the menu labels. I'm on a Mac :apple:)
- [ ] Edit 2.2 and 2.3 in atom/flight-manual.atom.io to reflect new functions

### CODE REVIEW

File: src/cursor.coffee [hunk 1]
```diff
@@ -311,8 +311,29 @@ class Cursor extends Model
...
+    #
+    firstCharacterColumn = null
+    @editor.scanInBufferRange /\S/, bufferLineBufferRange, ({range, stop}) ->
+      firstCharacterColumn = range.start.column
+      stop()
+
+    if firstCharacterColumn? and firstCharacterColumn isnt @getBufferColumn()
+      targetBufferColumn = firstCharacterColumn
+    else
+      targetBufferColumn = bufferLineBufferRange.start.column
```
Comment (nathansobo) [inline src/cursor.coffee]: Could this just be 0 or am I missing something?

**Comment by nathansobo** [comment]
⚡️ Nice work. This looks close. I suggest copying the existing test for moving to the first char of a screen line and modifying it. Make sure that deleting any piece of logic in your new method causes your test to fail.

File: src/cursor.coffee (hunk 1 above)
Comment (kylesimmonds) [inline src/cursor.coffee]: @nathansobo it probably is. I modified that function from the original, which is now `moveToFirstCharacterOfScreenLine`. I'll try it locally and update it if it works.

Comment (kylesimmonds) [inline src/cursor.coffee]: Updated

**Comment by kylesimmonds** [comment]
Updated checklist at the beginning of this PR to include updating the Flight Manual. Even though it's not in the scope of this project, I don't want to forget it.

**Comment by sadick254** [comment]
Hey @cinnamonkale. Apologies for not looking into this on time. I know this PR has been here for a very long time, during that time Atom has seen a number of changes. Due to the changes, this PR has now become outdated. Is it possible for you to resolve the conflicts?

**Comment by cinnamonkale** [comment]
@sadick254 I'd just close the issue unless you or someone close to the project is willing to merge it in. It might still be a relevant addition to Atom but I'm too far removed from it now to finish it

**Comment by sadick254** [comment]
@cinnamonkale thanks for responding. At this point let me just close it. I see this PR being useful in future when we have the bandwidth to fix the `select-to-end-of-line-behavior`. Thank you for your contribution.

**Pull request closed without merging**